contracts/utils/structs/DoubleEndedQueue.sol

```diff
@@ -78,12 +65,12 @@ library DoubleEndedQueue {
     /**
      * @dev Inserts an item at the beginning of the queue.
      *
-     * Reverts with {QueueFull} if the queue is full.
+     * Reverts with {RESOURCE_ERROR} panic code if the queue is full.
      */
     function pushFront(Bytes32Deque storage deque, bytes32 value) internal {
         unchecked {
             uint128 frontIndex = deque._begin - 1;
-            if (frontIndex == deque._end) revert QueueFull();
+            if (frontIndex == deque._end) Panic.panic(Panic.RESOURCE_ERROR);
```

Comment:
Not sure if this is the right fit because it indicates a resource error in memory. This is in storage.

I'd say using `Panic.ARRAY_OUT_OF_BOUNDS` would be a better fit, what do you think?

Reply:
This is not an access that falls beyond the limit of the array, this is a fundamental limitation. Ressources available are limited and this is trying to use more than is available.

Note that `RESOURCE_ERROR` is documented as
> resource error (too large allocation or too large array)

I believe "too large array" is pretty much the problem here.

Also, note that for this error to happen, someone would have to "push" more elements in the array than there are atoms in the observable universe. This is NEVER going to happen, so no need to worry to much.

Reply:
> This is NEVER going to happen, so no need to worry to much.

lmao yes I'm worrying too much.

> This is not an access that falls beyond the limit of the array, this is a fundamental limitation of the structure.

I still think the line is blur. Imo this is because the array **is bounded** to the 2**256 elements.

I'm solving since I'm fine either since it shouldn't happen.